client/modules/App/components/Overlay.jsx

```diff
@@ -9,10 +9,39 @@ class Overlay extends React.Component {
   constructor(props) {
     super(props);
     this.close = this.close.bind(this);
+    this.handleClick = this.handleClick.bind(this);
+    this.handleClickOutside = this.handleClickOutside.bind(this);
+    this.keyPressHandle = this.keyPressHandle.bind(this);
+  }
+
+  componentWillMount() {
+    document.addEventListener('mousedown', this.handleClick, false);
   }
 
   componentDidMount() {
-    this.overlay.focus();
+    this.node.focus();
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('mousedown', this.handleClick, false);
+  }
+
+  handleClick(e) {
+    if (this.node.contains(e.target)) {
+      return;
+    }
+
+    this.handleClickOutside();
+  }
+
+  handleClickOutside() {
+    this.close();
+  }
+
+  keyPressHandle(e) {
+    if (e.keyCode === 27) {
```

Comment:
could you add a comment here that 27 == escape key?

Reply:
sure